contracts/Fees.sol

```diff
@@ -28,43 +29,49 @@ abstract contract Fees is IFees, Owned {
         controllerGasLimit = _controllerGasLimit;
     }
 
-    function _fetchProtocolFees(PoolKey memory key)
-        internal
-        view
-        returns (uint8 protocolSwapFee, uint8 protocolWithdrawFee)
-    {
+    function _fetchProtocolFees(PoolKey memory key) internal view returns (uint24 protocolFees) {
+        uint16 protocolSwapFee;
+        uint16 protocolWithdrawFee;
         if (address(protocolFeeController) != address(0)) {
             // note that EIP-150 mandates that calls requesting more than 63/64ths of remaining gas
             // will be allotted no more than this amount, so controllerGasLimit must be set with this
             // in mind.
             if (gasleft() < controllerGasLimit) revert ProtocolFeeCannotBeFetched();
             try protocolFeeController.protocolFeesForPool{gas: controllerGasLimit}(key) returns (
-                uint8 updatedProtocolSwapFee, uint8 updatedProtocolWithdrawFee
+                uint16 updatedProtocolSwapFee, uint16 updatedProtocolWithdrawFee
             ) {
                 protocolSwapFee = updatedProtocolSwapFee;
                 protocolWithdrawFee = updatedProtocolWithdrawFee;
-            } catch {}
 
+                if (protocolSwapFee > MAX_FEE_DENOMINATOR) revert FeeDenominatorOutOfBounds(protocolSwapFee);
+                if (protocolWithdrawFee > MAX_FEE_DENOMINATOR) revert FeeDenominatorOutOfBounds(protocolWithdrawFee);
+
+                protocolFees = (uint24(protocolSwapFee) << 12) | protocolWithdrawFee;
+            } catch {}
             _checkProtocolFee(protocolSwapFee);
             _checkProtocolFee(protocolWithdrawFee);
         }
     }
 
     /// @notice There is no cap on the hook fee, but it is specified as a percentage taken on the amount after the protocol fee is applied, if there is a protocol fee.
-    function _fetchHookFees(PoolKey memory key) internal view returns (uint8 hookSwapFee, uint8 hookWithdrawFee) {
+    function _fetchHookFees(PoolKey memory key) internal view returns (uint24 hookFees) {
         if (key.fee.hasHookSwapFee()) {
-            hookSwapFee = IHookFeeManager(address(key.hooks)).getHookSwapFee(key);
+            uint16 swapFee = IHookFeeManager(address(key.hooks)).getHookSwapFee(key);
```

Comment:
What do you think about doing something like this? Allowing the integrator to return a full uint24 and then just making sure each 12 bits are only set if there is a fee set.

```
uint24 hookFee = IHookFeeManager(address(key.hooks)).getHookFee(key);
uint24 swapFeeMask = key.fee.hasHookSwapFee() ?  0xFFF : 0
uint24 withdrawFeeMask = key.fee.hasHookWithdrawFee() ? 0xFFF : 0
uint24 fullFeeMask = swapFeeMask << 12 | withdrawFeeMask
hookFee = hookFee & fullFeeMask
```

Reply:
And then I think for the future it could be nice to have a library to handle parsing uint24s to uint16 fees and vice versa but that can be separate from this PR

Reply:
If you all like this suggestion, would then probably suggest a similar interface for setting protocol fees so just having it return the uint24

Reply:
Yeah I'm a fan of this. I feel that having unintuitive UX in the pool is OK given its not meant to be directly interacted with by non-technical folk. 

When writing the parsing library you mentioned, could be nice to use custom Value Types... but that can be discussed separately.